src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -355,6 +351,23 @@ public AgentSettings LoadSettings()
             var serviceControlManager = HostContext.GetService<ILinuxServiceControlManager>();
             serviceControlManager.GenerateScripts(settings);
 #endif
+
+            // Get and apply Tags in case agent is configured against Machine Group
+            bool needToAddTags = agentProvider.GetAddTagsRequired(command);
```

Comment:
Is this for below experience?
`Do you want to add tags? <AddTagsRequired>`
`Enter Tags: <Tags>`
If yes, is it required? why don't we directly go with Tags?